### DISCUSSION TITLE
util.enlivenObjects is not working properly on v6.3.0, but it is fine in v5.4.0

### DISCUSSION THREAD

**Discussion by famasboy888** [fabricjs/fabric.js#10100] [General]
**CheckList**

- [x]  I agree to follow this project's Code of Conduct *
- [x]   I have read and followed the [Contributing Guide](https://github.com/fabricjs/fabric.js/blob/master/CONTRIBUTING.md) *
- [x]   I have read and followed the [Issue Tracker Guide](https://github.com/fabricjs/fabric.js/blob/master/CONTRIBUTING.md#%EF%B8%8F-issue-tracker) *
- [x]   I have searched and referenced existing issues and discussions *
- [x]   I am filing a BUG report. *
- [x]   I have managed to reproduce the bug after upgrading to the latest version *
- [x]   I have created an accurate and minimal reproduction *

**Version**
6.3.0

**In What environments are you experiencing the problem?**
- chrome, edge
- ReactJS

**Node Version (if applicable)**
None

**Link To Reproduction**
https://codesandbox.io/p/sandbox/upbeat-heisenberg-npkzng?file=%2Fsrc%2FApp.js

**Steps To Reproduce**
1. use latest fabric v6.3.0 via npm
2. initialize canvas element
3. add dummy shape and re-render canvas while using **enlivenObjects**
4. call cleanup function in useEffect hook

**Expected Behavior**

I expect v6.3.0 to be able to:
- add objects using **enlivenObjects**
- behave the same way like v5.4.0 as shown below 
- to console.log() something inside the callback function

**_**v5.4.0 (Working):**_** 
https://codesandbox.io/p/sandbox/smoosh-fabricv5-4-0-j3x36l?file=%2Fsrc%2FApp.js

<img width="30%" height="30%" src="https://github.com/user-attachments/assets/369dbec0-c026-43aa-9d2e-123e5cbbf819">
<img width="30%" height="30%" src="https://github.com/user-attachments/assets/8767f82d-8b43-43ac-9a39-b359df7dc43b">

```js
export default function App() {
  useEffect(() => {
    //Initialize canvas
    const canvas = new fabric.Canvas("canvas", {
      preserveObjectStacking: true,
    });

    //Load a dummy shape (Blue rect)
    const dummyShape = [
      {
        type: "Rect",
        left: 100,
        top: 100,
        width: 200,
        height: 200,
        fill: "blue",
      },
    ];

    fabric.util.enlivenObjects(dummyShape, (objs) => {
      console.log("Print from enlivenObjects", objs); //Print from enlivenObjects fabric v5.4.0
      objs.forEach((item) => {
        canvas.add(item);
      });
      canvas.renderAll();
    });

    //Clean up function
    return () => {
      canvas.dispose();
    };
  }, []);

  return <canvas id="canvas" width="1000" height="1000"></canvas>;
}
```

**Actual Behavior**

v6.3.0 is not able to:
- add objects using **enlivenObjects**
- to console.log() something inside the callback function

_Note: The code in both links are very identical, only the fabric version is different._

**_**v6.3.0 (Not Working):**_** 
https://codesandbox.io/p/sandbox/upbeat-heisenberg-npkzng?file=%2Fsrc%2FApp.js

<img width="30%" height="30%" src="https://github.com/user-attachments/assets/ddf9f0eb-8fcb-4e3a-80bd-b4a98c8b5df7">
<img width="30%" height="30%" src="https://github.com/user-attachments/assets/594f9ab3-0ac7-4708-98f7-d31d24f33705">

```js
export default function App() {
  useEffect(() => {
    //Initialize canvas
    const canvas = new Canvas("canvas", {
      preserveObjectStacking: true,
    });

    //Load a dummy shape (Blue rect)
    const dummyShape = [
      {
        type: "Rect",
        left: 100,
        top: 100,
        width: 200,
        height: 200,
        fill: "blue",
      },
    ];

    util.enlivenObjects(dummyShape, (objs) => {
      console.log("Print from enlivenObjects", objs); //Print from enlivenObjects fabric v6.3.0
      objs.forEach((item) => {
        canvas.add(item);
      });
      canvas.renderAll();
    });

    //Clean up function
    return () => {
      canvas.dispose();
    };
  }, []);

  return <canvas id="canvas" width="1000" height="1000"></canvas>;
}
```
**Error Message & Stack Trace**
None

**Comment by famasboy888** [comment]
Please disregard. I have found the answer in here for the usage of v6.3.0: https://fabricjs.github.io/api/namespaces/util/functions/enlivenobjects/

**Reply by asturur** [reply]
Please also read this
https://fabricjs.github.io/docs/upgrading/upgrading-to-fabric-60/

**Reply by Amrjigga** [reply]
you're a life saver
